pkg/services/sqlstore/datasource.go

```diff
@@ -129,6 +129,40 @@ func (ss *SQLStore) DeleteDataSource(ctx context.Context, cmd *models.DeleteData
 	})
 }
 
+// DeleteDataSourceSecrets removes a datasource secure_json_data by org_id as well as either uid (preferred), id, or name.
+func (ss *SQLStore) DeleteDataSourceSecrets(ctx context.Context, cmd *models.DeleteDataSourceSecretsCommand) error {
+	return ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
+		dsQuery := &models.GetDataSourceQuery{Id: cmd.ID, Uid: cmd.UID, Name: cmd.Name, OrgId: cmd.OrgID}
+		errGettingDS := ss.getDataSource(ctx, dsQuery, sess)
+
+		if errGettingDS != nil && !errors.Is(errGettingDS, models.ErrDataSourceNotFound) {
```

Comment:
I think we should handle also the `errors.Is(errGettingDS, models.ErrDataSourceNotFound)` case.
But still I don't know if we want to execute the extra query just for checking the existence of the data source. When we update the data source [we just check the affected rows](https://github.com/grafana/grafana/blob/ab4171770e7fa78edc857e7dda20426382f9e367/pkg/services/sqlstore/datasource.go#L260) of the update statement.